packages/core/src/stateUtils.ts

```diff
@@ -578,10 +581,12 @@ export function getStateNode(
       `Unable to retrieve child state '${stateKey}' from '${stateNode.id}'; no child states exist.`
     );
   }
-  const result = stateNode.states[stateKey];
+  // replace escaped period ('\.') with just a period
+  const resolvedStateKey = stateKey.replace(/\\\./g, '.');
+  const result = stateNode.states[resolvedStateKey];
```

Comment:
There is no test case that would fail if I revert this change:
```suggestion
  const result = stateNode.states[stateKey];
```